Tests/OTelTests/Logging/OTelBatchLogRecordProcessorTests.swift

```diff
@@ -0,0 +1,178 @@
+        let batchProcessor = OTelBatchLogRecordProcessor(
+            exporter: exporter,
+            configuration: OTelBatchLogRecordProcessorConfiguration(
+                environment: .detected(),
+                maximumQueueSize: 100,
+                scheduleDelay: .milliseconds(10)
+            ),
+            clock: clock
+        )
+
+        await withThrowingTaskGroup(of: Void.self) { taskGroup in
+            taskGroup.addTask(operation: batchProcessor.run)
+            var iterator = exporter.didExportBatch.makeAsyncIterator()
+
+            let logHandler = OTelLogHandler(
+                processor: batchProcessor,
+                logLevel: .debug,
+                resource: resource
+            )
+            let logger = Logger(label: "Test", logHandler)
+
+            for i in 1...4 {
+                logger.info("\(i)")
+            }
+
+            // TODO: Records are emitted asynchronously, so checking this without delay
+            // is not representative
+            XCTAssertEqual(exporter.records.count, 0)
+
+            await sleeps.next()
+            clock.advance(by: .milliseconds(10))
+            await sleeps.next()
+
+            // Records should update after delay
+            await iterator.next()
+            XCTAssertEqual(exporter.records.count, 4)
+
+            taskGroup.cancelAll()
+        }
+    }
+
+    func testBatchLogProcessorForceFlushEmitsEarly() async throws {
+        let exporter = OTelInMemoryLogRecordExporter()
+        let clock = TestClock()
+        var sleeps = clock.sleepCalls.makeAsyncIterator()
+        let batchProcessor = OTelBatchLogRecordProcessor(
+            exporter: exporter,
+            configuration: OTelBatchLogRecordProcessorConfiguration(
+                environment: .detected(),
+                maximumQueueSize: 5,
+                scheduleDelay: .seconds(60) // Should never trigger
+            ),
+            clock: clock
+        )
+
+        try await withThrowingTaskGroup(of: Void.self) { taskGroup in
+            taskGroup.addTask(operation: batchProcessor.run)
+
+            let logHandler = OTelLogHandler(
+                processor: batchProcessor,
+                logLevel: .debug,
+                resource: resource
+            )
+            let logger = Logger(label: "Test", logHandler)
+
+            for i in 1...4 {
+                logger.info("\(i)")
+            }
+
+            await sleeps.next()
+
+            // TODO: Records are emitted asynchronously, so checking this without delay
+            // is not representative
+            XCTAssertEqual(exporter.records.count, 0)
+
+            try await batchProcessor.forceFlush()
+            XCTAssertEqual(exporter.records.count, 4)
+
+            taskGroup.cancelAll()
+        }
+    }
+
+    func testBatchLogProcessorCancelsSlowExports() async throws {
+        let exporter = OTelSlowLogRecordExporter(delay: .milliseconds(50))
+        let batchProcessor = OTelBatchLogRecordProcessor(
```

Comment:
Can we use a `TestClock` to control time for both the waiting processor and the slow exporter?